src/aap_eda/core/models/activation.py

```diff
@@ -193,22 +195,36 @@ class Meta:
         default=ActivationStatus.PENDING,
     )
     git_hash = models.TextField(null=False, default="")
-    activation = models.ForeignKey("Activation", on_delete=models.CASCADE)
+
+    # TODO: remove this field
+    activation = models.ForeignKey(
+        "Activation", on_delete=models.CASCADE, null=True
+    )
     started_at = models.DateTimeField(auto_now_add=True, null=False)
     updated_at = models.DateTimeField(null=True)
     ended_at = models.DateTimeField(null=True)
     activation_pod_id = models.TextField(null=True)
     status_message = models.TextField(null=True, default=None)
     log_read_at = models.DateTimeField(null=True)
+    parent_id = models.TextField(null=False)
+    parent_fqcn = models.TextField(null=False)
+
+    def get_parent(self):
+        module_name, class_name = self.parent_fqcn.rsplit(".", 1)
```

Comment:
This is can be done by `from django.utils.module_loading import import_string`
```
parent_model = import_string(self.parent_fqcn)
```

Reply:
do this only if `self.parent_fqcn` is not None, otherwise return `self.activation`

Reply:
@Alex-Izquierdo This is still a valid function in Django 5.0
https://docs.djangoproject.com/en/5.0/ref/utils/#django.utils.module_loading.import_string